R/augsynth.R

```diff
@@ -1,323 +1,74 @@
-    ## straight from lm
-    cat("\nCall:\n", paste(deparse(augsynth$call), sep="\n", collapse="\n"), "\n\n", sep="")
-
-    ## print att estimates
-    att_post <- colMeans(augsynth$data$y[augsynth$data$trt == 1,,drop=F]) -
-        predict(augsynth)
-
-    cat(paste("Average ATT Estimate: ",
-              format(round(mean(att_post),3), nsmall = 3), "\n\n", sep=""))
-}
-
-
-#' Plot function for augsynth
-#' @param se Whether to plot standard errors
-#' @param jackknife Whether to use jackknife or weighted SEs
-#' @export
-plot.augsynth <- function(augsynth, se = T, jackknife = T) {
-    plot(summary(augsynth, jackknife = jackknife), se = se)
-}
-
-
-#' Summary function for augsynth
-#' @param jackknife Whether to use jackknife or weighted SEs
-#' @export
-summary.augsynth <- function(augsynth, jackknife = T) {
-
-    summ <- list()
-
-    # ## post treatment estimate
-    # att_post <- colMeans(augsynth$data$y[augsynth$data$trt == 1,,drop=F]) -
-    #     predict(augsynth)
-
-    # ## pre treatment estimate
-    # att_pre <- colMeans(augsynth$data$X[augsynth$data$trt == 1,,drop=F]) -
-    #     t(augsynth$data$X[augsynth$data$trt==0,,drop=F]) %*% augsynth$weights
-    att_est <- predict(augsynth, att = T)
-    att <- data.frame(Time = augsynth$data$time,
-                        Estimate = att_est)
-
-
-    if(augsynth$progfunc == "Ridge" |
-       augsynth$progfunc == "None" & augsynth$scm) {
-        ridge <- augsynth$progfunc == "Ridge"
-        scm <- augsynth$scm
-
-        ## get standard errors
-        synth_data <- format_synth(augsynth$data$X, augsynth$data$trt,
-                                   augsynth$data$y)
-
-        if(jackknife) {
-            att_se <- jackknife_se_ridgeaug(augsynth$data, synth_data,
-                                            augsynth$data$Z, augsynth$lambda,
-                                            ridge, scm, augsynth$fixedeff)
-        } else {
-            att_se <- loo_se_ridgeaug(augsynth$data, synth_data, 
-                                      augsynth$data$Z,
-                                      augsynth$lambda,
-                                      ridge, scm)
-        }
-
-        att$Std.Error <- att_se$se
-
-        summ$att <- att
-        summ$sigma <- att_se$sigma
-    } else {
-        ## no standard errors
-        att$Std.Error <- NA
-        summ$att <- att
-        summ$sigma <- NA
+#' 
+augsynth <- function(form, unit, time, data, t_int=NULL, ...) {
+
+  call_name <- match.call()
+  
+  form <- Formula::Formula(form)
+  unit_quosure <- enquo(unit)
+  time_quosure <- enquo(time)
+  
+  ## format data
+  outcome <- terms(formula(form, rhs=1))[[2]]
+  trt <- terms(formula(form, rhs=1))[[3]]
+  wide <- format_data_stag(outcome, trt, unit_quosure, time_quosure, data=data)
+
+  trt_year = -1
+  multi = F
+  for (u in wide$trt) {
+    if (is.finite(u) && u != trt_year) {
+      if (trt_year == -1) {
+        trt_year = u
+      } else {
+        multi = T
+        break
+      }
```

Comment:
I think this can be more simple to just check if `wide$trt` contains more than 2 unique values. Something like `length(unique(wide$trt)) > 2`

Reply:
Can we assume that there will always be `Inf` as a value in `wide$trt` so that `length(unique(wide$trt)) > 2` is the same as having 2 unique non-infinite values?

Reply:
Yeah, if `wide$trt` doesn't have at least two unique values then the input data is messed up (e.g. all the units are untreated or treated). We might actually want to put in a check for that in `format.R`